### DISCUSSION TITLE
Warning in calculating mean radiation temperature and shadow

### DISCUSSION THREAD

**Discussion by land333land** [UMEP-dev/UMEP#548] [Q&A]
I have prepared all the pre-data for calculating shade and mean radiant temperature, but when I put them in I find that it did not start running and suggested the following error. In fact, when I was calculating sky, it was running fine. I used the test data provided and found that it doesn't run either.

WARNING    Traceback (most recent call last):
              File "C:\Users/YLS/AppData/Roaming/QGIS/QGIS3\profiles\ccfdd/python/plugins\UMEP\SOLWEIG\solweig.py", line 493, in start_progress
              if np.where(self.lcgrid == ndlc)[0] > 0:
             ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()

**Comment by biglimp** [comment]
Could you attach your land cover grid for us to have a look at. @nilswallenberg , didnt you do something to avoid nodata issues here?

**Comment by land333land** [comment]
Hello, 


The attachment contains the land use data that I used. 


Thank you in advance for your help.
















At 2023-09-11 14:38:53, "biglimp" ***@***.***> wrote:

Could you attach your land cover grid for us to have a look at. @nilswallenberg , didnt you do something to avoid nodata issues here?

**Reply by biglimp** [reply]
Sorry, I cannot see your attachment. Can you try again?

**Comment by land333land** [comment]
Hello, I've uploaded the attachment again.
















At 2023-09-12 15:40:39, "biglimp" ***@***.***> wrote:

Sorry, I cannot see your attachment. Can you try again?

**Reply by biglimp** [reply]
Strange, I cannot see any attachment. Do you see my attached textfile? I drag and dropped.
[Testing.txt](https://github.com/UMEP-dev/UMEP/files/12584877/Testing.txt)

**Comment by land333land** [comment]
How about this time?













At 2023-09-12 17:40:19, "biglimp" ***@***.***> wrote:

Strange, I cannot see any attachment. Do you see my attached textfile? I drag and dropped.
Testing.txt

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

从网易163邮箱发来的超大附件推荐客户端极速下载
LULC_UMEP10.tif (35.86M, 2023年9月27日 21:37 到期)
下载
LULC_UMEP10.tif.aux.xml (407B, 2023年9月27日 21:37 到期)
在线预览 | 下载
LULC_UMEP10.tif.ovr (419.43K, 2023年9月27日 21:37 到期)
下载

**Reply by biglimp** [reply]
No, sorry. Can you send it in another way, e.g. via file transfer?

**Reply by land333land** [reply]
[Land cover_912.zip](https://github.com/UMEP-dev/UMEP/files/12587379/Land.cover_912.zip)
[Land cover.zip](https://github.com/UMEP-dev/UMEP/files/12587381/Land.cover.zip)

**Comment by biglimp** [comment]
You have found a bug. We will fix this in upcoming version. In the meantime, could you please try out SOLWEIG via **UMEP for Processing** instead. I think it should work from there.

If you still like to use menu-based UMEP, change line 493 in solweig.py to:
`if np.where(self.lcgrid == ndlc)[0].shape[0] > 0: # fixed issue from Discussions #548`

OR

download the plugin driectly from our repository when I have fixed this issue.
